sympy/holonomic/holonomic.py

```diff
@@ -2109,12 +2109,26 @@ def expr_to_holonomic(func, x=None, initcond=True, x0=0, lenics=None, domain=QQ,
     meijerint._rewrite1, _convert_poly_rat_alg, _create_table
     """
     func = sympify(func)
+    syms = func.free_symbols
+
     if not x:
-        syms = func.free_symbols
         if len(syms) == 1:
             x= syms.pop()
         else:
             raise ValueError("Specify the variable for the function")
+    else:
+        if len(syms) != 0:
+            syms.remove(x)
+
+    extra_syms = [i for i in syms]
+
+    if domain == None:
+        if func.has(Float):
+            domain = RR
+        else:
+            domain = QQ
+        if len(extra_syms) != 0:
+            domain = domain[extra_syms]
```

Comment:
This should probably be made a field (using `get_field()`). The extra symbols may also occur in denominators.